config/hostid/hostid.go

```diff
@@ -50,34 +64,134 @@ type Configuration struct {
 
 	// EnvVarName is the environment specified host ID if using environment host ID resolver.
 	EnvVarName *string `yaml:"envVarName"`
+
+	// File is the kv file config.
+	File *FileConfig `yaml:"file"`
 }
 
-// Resolve returns the resolved host ID given the configuration.
-func (c Configuration) Resolve() (string, error) {
+// FileConfig contains the info needed to construct a FileResolver.
+type FileConfig struct {
+	// Path of the file containing the KV config.
+	Path string `yaml:"path"`
+
+	// Timeout to wait for the file to be non-empty.
+	Timeout time.Duration `yaml:"timeout"`
```

Comment:
nit: Maybe make this a pointer type and use a default Timeout if not specified?  Of say 10s?